rosa-hypershift/rosa-hosted-wrapper.py

```diff
@@ -215,14 +353,66 @@ def _download_kubeconfig(ocm_cmnd, cluster_id, my_path):
+    return_data = {}
+    logging.info('Creating cluster-admin user on cluster %s (100 retries allowed)' % cluster_name)
+    kubeconfig_cmd = [rosa_cmnd,  "create", "admin", "-c", cluster_name, "-o", "json"]
+    logging.debug(kubeconfig_cmd)
+    cluster_admin_create_time = int(time.time())
+    for trying in range(1, 101):
+        process = subprocess.Popen(kubeconfig_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=my_path, universal_newlines=True)
+        stdout, stderr = process.communicate()
+        if process.returncode != 0:
+            logging.warning('Try: %d. Failed to create cluster-admin user on %s with this stdout/stderr:' % (trying, cluster_name))
+            logging.warning(stdout)
+            logging.warning(stderr)
+            logging.error('Try: %d. Waiting 5 seconds for the next try on %s' % (trying, cluster_name))
+            time.sleep(5)
+        else:
+            return_data['cluster-admin-create'] = int(time.time()) - cluster_admin_create_time
+            logging.info('Trying to login on cluster %s (100 retries allowed, 5s timeout on oc command)' % cluster_name)
+            oc_login_time = int(time.time())
+            for trying2 in range(1, 101):
+                oc_login_cmnd = ["oc", "login", json.loads(stdout)['api_url'], "--username", json.loads(stdout)['username'], "--password", json.loads(stdout)['password'], "--kubeconfig", my_path + "/kubeconfig", "--insecure-skip-tls-verify=true", "--request-timeout=30s"]
+                logging.debug(oc_login_cmnd)
+                process_oc_login = subprocess.Popen(oc_login_cmnd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=my_path, universal_newlines=True)
+                stdout_oc_login, stderr_oc_login = process_oc_login.communicate()
+                if process_oc_login.returncode != 0:
+                    logging.warning('Try: %d. Failed to login on cluster %s with cluster-admin user with this stdout/stderr:' % (trying2, cluster_name))
+                    logging.warning(stdout_oc_login)
+                    logging.warning(stderr_oc_login)
+                    logging.warning('Try: %d. Waiting 5 seconds for the next try on %s' % (trying2, cluster_name))
+                    time.sleep(5)
+                else:
+                    logging.info("Try: %d. Login succesfull on cluster %s, checking if kubeconfig has been downloaded" % (trying2, cluster_name))
+                    with open(my_path + "/kubeconfig", "r") as kubeconfig_file:
+                        kubeconfig = kubeconfig_file.readlines()
+                        for line in kubeconfig:
+                            if json.loads(stdout)['api_url'] in line:
+                                logging.info("Try %d. Valid kubeconfig downloaded for %s" % (trying2, cluster_name))
+                                return_data['kubeconfig'] = my_path + "/kubeconfig"
+                                return_data['cluster-admin-login'] = int(time.time()) - oc_login_time
```

Comment:
Just curious, `cluster-admin-login` time should be the interval it takes for first login isn't it ? If so, should it be calculated after the Login successful(L387), instead of waiting on kubeconfig file download.

Reply:
I can move it, but waiting for checking the kubeconfig is not adding even a second to the count and, if kubeconfig is wrong, it will retry, so we could be on the last retry, login correct but wrong kubeconfig, if so. we could be giving a time without having the access

Reply:
@morenod but what about the situation yesterday? When the account was created and we pulled a kubeconfig but it didnt actually have priv's to do anything. Is this covered at all?

Reply:
no, it is not covered. I will add some command to check if it works, like getting the list of projects or something

Reply:
add oc adm command to verify than we have admin rights with that user

Reply:
@morenod can we index this additional timing as well?